### PR TITLE
provides initial low-level integration with rsocket

### PR DESCRIPTION
**Pull request by OlegDokuka**

### CODE REVIEW

File: spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/RSocketAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package org.springframework.cloud.sleuth.autoconfig.instrument.rsocket;
```
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/RSocketAutoConfiguration.java:1]: Missing copyrights

File: spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/SleuthRSocketProperties.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * Sleuth RSocket settings.
+ *
+ * @author Oleh Dokuka
+ * @since 3.1.0
+ */
+@ConfigurationProperties("spring.sleuth.rsocket")
+public class SleuthRSocketProperties {
+
+	/**
+	 * When true enables instrumentation for reactor.
```
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/SleuthRSocketProperties.java:31]: ...for RSocket

File: spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/RSocketAutoConfiguration.java (hunk 1 above)
Comment (OlegDokuka) [reply spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/RSocketAutoConfiguration.java:17]: fixed

File: spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/SleuthRSocketProperties.java (hunk 2 above)
Comment (OlegDokuka) [reply spring-cloud-sleuth-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/instrument/rsocket/SleuthRSocketProperties.java]: fixed

**Comment by marcingrzejszczak** [comment]
moved to https://github.com/spring-cloud/spring-cloud-sleuth/pull/1929

**Pull request closed without merging**